_posts/2019-06-07-python-datascience-with-openfaas.md

```diff
@@ -0,0 +1,267 @@
+
+## The `pydatascience template`
+The [pydatascience template][pydata-template] showcases a couple of ideas that anyone can leverage in their own functions and templates:
+
+1. using the [Conda package manager][conda-homepage]
+2. setting up a non-root user Python environment
+3. [multi-module Python function][multifile-blog-post]
+4. using HTTP mode in [of-watchdog][of-watchdog-homepage] to load an asset (in this case a model) into memory during startup
+
+## The name classifier function
+
+In this post, we share an example [name classifier][name-classifier-repo].  It is a relatively simple function that accepts a `name` and then attempts to guess the nationality of that name, returning the top three guesses.
+
+```sh
+echo "Lucas" | faas-cli invoke classify
+[
+  ["(-1.4300)","Irish"],
+  ["(-1.6112)","Italian"],
+  ["(-2.0314)","Dutch"]
+]
+```
+
+You can try this in your own OpenFaaS cluster using
+
+```sh
+faas-cli deploy --image=theaxer/classify:latest --name=classify
+```
+
+Caution, I did not say it would be a great guesser. This model was built on a relatively small data set and is intended to be a simple demonstration. And remember, with all things data science, "with great power comes great responsibility".
+
+## Understanding the template
+
+### Why Conda
+I have recently started to use Conda for many of my own experiments with Python, especially when they involve Pandas and Numpy, because the install is very fast and environment management is fairly easy.  For function development and build, the speed is really nice. For this project it was also the easiest way to [install PyTorch][pytorch-install].  Finally, Conda also supports non-Python packages, e.g. curl, yarn-js, and zeromq are installable via Conda.
+
+### Setting up a non-root function
+If you spend much time with Docker containers, you will hear people talk about using non-root images. These images tend to be much safer to deploy. Fortunately, OpenFaaS [makes it easy to enforce][of-force-non-root], even when the original image is not non-root by default.  When we build new templates, it is important that we consider non-root deployments as the default. For Python environments, Conda made this very easy.
```

Comment:
`If you spend much time with Docker containers, you will hear people talk about using non-root images.`

To:

A best practice for deploying Docker images is to make sure that the user running your code is not privileged, this is also known as "non-root".